### PR TITLE
Vid 28 file

### PR DESCRIPTION
**Pull request by Mantas330BJ**

### CODE REVIEW

**Review by gzili — APPROVED** [review]
LGTM

File: backend/src/main/java/com/videolibrary/backend/BackendApplication.java [hunk 1]
```diff
@@ -1,9 +1,12 @@
 package com.videolibrary.backend;
 
+import com.videolibrary.backend.property.StorageProperties;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
 
 @SpringBootApplication
+@EnableConfigurationProperties(StorageProperties.class)
```
Comment (edvinasstaupas) [inline backend/src/main/java/com/videolibrary/backend/BackendApplication.java:9]: why is this needed? I think if you annotate `StorageProperties.class` with @Configuration, this annotation is not required.

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.UUID;
+
+@RestController
+@RequestMapping("files")
+@RequiredArgsConstructor
+public class FileUploadController {
+    private final FileStorageService storageService;
+
+    @GetMapping("{filename}")
+    public ResponseEntity<Resource> serveFile(@PathVariable String filename, @RequestParam("type") FileType type) {
```
Comment (edvinasstaupas) [inline backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:26]: Could you explain FileType in this context? As i understand, there is no difference between type files. Why not store them in the same folder, they are unique by names.

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("files")
+@RequiredArgsConstructor
+public class FileUploadController {
+    private final FileStorageService storageService;
+
+    @GetMapping("{filename}")
```
Comment (edvinasstaupas) [inline backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:25]: You should add octet-stream as `produces` type

File: backend/src/main/resources/application.properties
Comment (edvinasstaupas) [inline backend/src/main/resources/application.properties:1]: `storage.rootStoragePath` not present.

File: backend/src/main/java/com/videolibrary/backend/BackendApplication.java (hunk 1 above)
Comment (Mantas330BJ) [reply backend/src/main/java/com/videolibrary/backend/BackendApplication.java:9]: Annotating as Configuration is also an option. It is just what I prefer

This article recommends several ways to configure configuration properties
https://www.baeldung.com/configuration-properties-in-spring-boot

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java (hunk 2 above)
Comment (Mantas330BJ) [reply backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:26]: Files could be stored in the same folder. I just thought it would be easier to navigate when thumbnails are separated from videos. Not really a strong argument

Comment (edvinasstaupas) [reply backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:26]: It adds a lot of complexity and i dont think its necessary. But if it works it works.

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java (hunk 3 above)
Comment (Mantas330BJ) [reply backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:25]: What would be the main reason for it?
I did this endpoint according to the example. Look at 'serveFile' method
https://spring.io/guides/gs/uploading-files/

File: backend/src/main/java/com/videolibrary/backend/BackendApplication.java (hunk 1 above)
Comment (edvinasstaupas) [reply backend/src/main/java/com/videolibrary/backend/BackendApplication.java:9]: The problem rises when there are tens or even more properties classes.

File: backend/src/main/resources/application.properties
Comment (Mantas330BJ) [reply backend/src/main/resources/application.properties:1]: Yes. I specify the default value in the field of the object

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java (hunk 2 above)
Comment (Mantas330BJ) [reply backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:26]: There might be different processing for thumbnails and videos in the future, so I think having a file type to make a disctinction makes some sense.

File: backend/src/main/resources/application.properties
Comment (edvinasstaupas) [reply backend/src/main/resources/application.properties:1]: I would say bad practice

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java (hunk 2 above)
Comment (Mantas330BJ) [reply backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:26]: Will delete for know

File: backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java (hunk 3 above)
Comment (edvinasstaupas) [reply backend/src/main/java/com/videolibrary/backend/controller/FileUploadController.java:25]: Data is usually transmitted with octet-stream. I dont think there is default setting for produces

File: backend/src/main/java/com/videolibrary/backend/BackendApplication.java (hunk 1 above)
Comment (Mantas330BJ) [reply backend/src/main/java/com/videolibrary/backend/BackendApplication.java:9]: Used autoconfiguration

File: backend/src/main/resources/application.properties
Comment (Mantas330BJ) [reply backend/src/main/resources/application.properties:1]: Is it bad to have defaults?

Comment (edvinasstaupas) [reply backend/src/main/resources/application.properties:1]: default values can be specified in properties as well

Comment (Mantas330BJ) [reply backend/src/main/resources/application.properties:1]: Extracted to application.yml
